components/Calendar.js

```diff
@@ -37,17 +36,16 @@ export default function Calendar({ emotions }) {
   };
 
   const getEmotionsForDay = (date) => {
-    return emotions.filter(($emotion) => {
-      const emotionDate = new Date($emotion.dateTime);
+    return emotions.filter((emotion) => {
+      const emotionDate = new Date(emotion.dateTime);
+
       return emotionDate.toDateString() === date.toDateString();
     });
   };
 
   const getColorByEmotionType = (type) => {
-    const $emotion = initialEmotionTypes.find(
-      (emotion) => emotion.emotionType === type
-    );
-    return $emotion ? $emotion.color : "var(--color-background)";
+    const emotionCol = emotions.find((emotion) => emotion.type.name === type);
```

Comment:
Es sollte eigentlich mit der _id hier gearbeitet werden, um die richtige emotion zu finden. Also gebt an die Funktion nicht `type` sondern `emotionId` und nutzt im find `emotion.type._id`. Nennt auch die Variable `emotionCol` um in `foundEmotion`.

Reply:
mit .type._id funktioniert es nicht, aus irgendeinem Grund. Mit .name schon.